### DISCUSSION TITLE
Adding SSL configuration through RabbitConnectionFactoryBean approach into the documentation

### DISCUSSION THREAD

**Discussion by kstojanovski-work** [spring-projects/spring-amqp#2471] [Q&A] [answered]
I was searching for a way, how to add the SSL properties path programmatically, and after a long search ended up creating RabbitConnectionFactoryBean.
i.e.
```
//    Possible solution for TLS / SSL
RabbitConnectionFactoryBean rabbitConnectionFactoryBean = new RabbitConnectionFactoryBean();
rabbitConnectionFactoryBean.setUseSSL(true);
//    file where the SSL properties are described in
//    https://docs.spring.io/spring-amqp/reference/html/#rabbitconnectionfactorybean-configuring-ssl
//    keyStore, trustStore, keyStore.passPhrase, trustStore.passPhrase
rabbitConnectionFactoryBean.setSslPropertiesLocation(new ClassPathResource("ssl.properties"));
CachingConnectionFactory cachingConnectionFactory = new CachingConnectionFactory(rabbitConnectionFactoryBean.getRabbitConnectionFactory());
```
Is this the correct way of doing this? And if yes, can this kind of code be a part of the documentation?

**Comment by artembilan** [comment, marked as answer]
Not sure why is the question. We indeed have similar sample in the doc: https://docs.spring.io/spring-amqp/reference/html/#rabbitconnectionfactorybean-configuring-ssl.

Although you might be interested in a Java variant.
If that, then let us know and we convert this discussion to an issue and proceed from there as a docs improvement.

**Reply by kstojanovski-work** [reply]
I see now the title **RabbitConnectionFactoryBean** and the bean in XML format but somehow I have not perceived it as what I needed. It is my bad! But it would be nice if there is also added the java variant in the documentation.
